src/test/regress/citus_tests/test/test_columnar.py

```diff
@@ -0,0 +1,113 @@
+import psycopg
+import pytest
+
+
+def test_freezing(coord):
+    coord.configure("vacuum_freeze_min_age = 50000")
+    coord.configure("vacuum_freeze_table_age = 50000")
+    coord.restart()
+
+    # create columnar table and insert simple data to verify the data survives
+    # a crash
+    coord.sql("CREATE TABLE test_row(i int)")
+    coord.sql("INSERT INTO test_row VALUES (1) ")
+    coord.sql(
+        "CREATE TABLE test_columnar_freeze(i int) USING columnar WITH(autovacuum_enabled=false)"
+    )
+    coord.sql("INSERT INTO test_columnar_freeze VALUES (1)")
+
+    for _ in range(0, 7):
+        with coord.cur() as cur:
+            for _ in range(0, 10_000):
+                cur.execute("UPDATE test_row SET i = i + 1")
+
+    frozen_age = coord.sql_value(
+        """
+        select age(relfrozenxid)
+        from pg_class where relname='test_columnar_freeze';
+    """
+    )
+
+    assert frozen_age > 70000, "columnar table was frozen"
+    coord.sql("VACUUM FREEZE test_columnar_freeze")
+
+    frozen_age = coord.sql_value(
+        """
+        select age(relfrozenxid)
+        from pg_class where relname='test_columnar_freeze';
+    """
+    )
+    assert frozen_age < 70000, "columnar table was not frozen"
+
+
+def test_recovery(coord):
+    coord.sql("CREATE TABLE t1 (a int, b text) USING columnar")
+    coord.sql(
+        "INSERT INTO t1 SELECT a, 'hello world ' || a FROM generate_series(1,1002) AS a"
+    )
+
+    # simulate crash
+    coord.stop("immediate")
+    coord.start()
+
+    row_count = coord.sql_value("SELECT count(*) FROM t1")
+    assert row_count == 1002, "columnar didn't recover data before crash correctly"
+
+    coord.sql("TRUNCATE t1")
+    # simulate crash
+    coord.stop("immediate")
+    coord.start()
+
+    row_count = coord.sql_value("SELECT count(*) FROM t1")
+    assert row_count == 0, "columnar didn't recover the truncate correctly"
+
+    with pytest.raises(
+        psycopg.OperationalError,
+        match="server closed the connection unexpectedly|consuming input failed: EOF detected",
+    ):
+        with coord.transaction() as cur:
+            cur.execute(
+                "INSERT INTO t1 SELECT a, 'hello world ' || a FROM generate_series(1,1003) AS a"
+            )
+            # simulate crash
+            coord.stop("immediate")
+
+    coord.start()
+
+    row_count = coord.sql_value("SELECT count(*) FROM t1")
+    assert row_count == 0, "columnar didn't recover uncommited transaction"
+
+    with pytest.raises(
```

Comment:
To on top of those `with pytest.raises()`  blocks, can we copy the comments from .pl files that details what we're testing, as in `# test crashing while having a prepared transaction`